tools/xarray_import_data/xarray_import_data.xml

```diff
@@ -0,0 +1,64 @@
+<tool id="xarray_import_data" name="Xarray Import Data" version="0.1.0" profile="22.05">
+    <description>
+        Import a dataset from an OPeNDAP URL and convert it to a local NetCDF file using xarray.
+    </description>
+    <requirements>
+        <requirement type="package" version="2025.4.0">xarray</requirement>
+        <requirement type="package" version="3">python</requirement>
+        <requirement type="package" version="1.6.0">netcdf4</requirement>
+    </requirements>
+    <command detect_errors="exit_code"><![CDATA[
+        python '$__tool_directory__/xarray_import_data.py' '$opendap_url' '$decode_times' '$decode_cf' '$output_dataset' 
+    ]]></command>
+    <inputs>
+        <param name="opendap_url" type="text" format="netcdf" label="OPeNDAP URL"
```

Comment:
what happens if a user puts a github/youtube whatever URL in here?

Reply:
Btw, should be easy to add support for s3 URLs ++ if needed.

Reply:
> Btw, should be easy to add support for s3 URLs ++ if needed.

I guess at this stage we should really explore the deferred-dataset use-case a bit more. If you like tell me where you got stuck in testing it.

Reply:
Yep, that is a good idea! When I tested deferred dataset it didn't pass the URL, but just an empty string, it was also a bit fiddly to get to the deferred option, but it would be nice having it easily working 👍

Reply:
Here is the schema for the validators: https://docs.galaxyproject.org/en/latest/dev/schema.html#tool-inputs-param-validator

Reply:
Thanks, yep added a small one, but the url can take quite many forms.